core/src/test/scala/org/apache/spark/sql/test/generator/TestDataGenerator.scala

```diff
@@ -0,0 +1,375 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.test.generator
+
+import com.pingcap.tikv.row.ObjectRowImpl
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.sql.test.generator.DataType._
+
+import scala.collection.mutable
+import scala.util.Random
+
+object TestDataGenerator {
+
+  type TiRow = com.pingcap.tikv.row.Row
+
+  val bits = List(BIT)
+  val booleans = List(BOOLEAN)
+  val integers = List(TINYINT, SMALLINT, MEDIUMINT, INT, BIGINT)
+  val decimals = List(DECIMAL)
+  val doubles = List(FLOAT, DOUBLE)
+
+  val timestamps = List(TIMESTAMP, DATETIME)
+  val dates = List(DATE)
+  val durations = List(TIME)
+  val years = List(YEAR)
+
+  val texts = List(TEXT, TINYTEXT, MEDIUMTEXT, LONGTEXT)
+  val strings = List(VARCHAR, CHAR)
+  val binaries = List(VARBINARY, BINARY)
+  val bytes = List(BLOB, TINYBLOB, MEDIUMBLOB, LONGBLOB)
+  val enums = List(ENUM)
+  val sets = List(SET)
+
+  val jsons = List(JSON)
+
+  // M indicates the number of digits
+  val withM: List[ReflectedDataType] = bits ::: integers ::: decimals ::: doubles
+  // D indicates the number of digits after the decimal point
+  val withD: List[ReflectedDataType] = decimals ::: doubles
+  // U indicates whether the datatype is unsigned
+  val withU: List[ReflectedDataType] = integers ::: decimals ::: doubles
+  // F indicates the fraction of date/datetime format
+  val withF: List[ReflectedDataType] = timestamps ::: durations
+
+  val numeric: List[ReflectedDataType] = bits ::: booleans ::: integers ::: decimals ::: doubles
+  val dateAndDateTime: List[ReflectedDataType] = timestamps ::: dates ::: durations ::: years
+
+  val stringAndBinaries: List[ReflectedDataType] = strings ::: binaries
+  // TODO: support enum and set
+  // val stringType: List[DataType] = texts ::: strings ::: binaries ::: enums ::: sets
+  val stringType: List[ReflectedDataType] = texts ::: strings ::: binaries ::: bytes
+  val varString: List[ReflectedDataType] = List(VARCHAR, VARBINARY)
+
+  val unsignedType: List[ReflectedDataType] = numeric
+
+  // TODO: support json
+  val allDataTypes: List[ReflectedDataType] = numeric ::: dateAndDateTime ::: stringType
+  // supported data types for generator
+  val supportedDataTypes: List[ReflectedDataType] = allDataTypes
+
+  // basic data types that represent each type category
+  val baseDataTypes: List[ReflectedDataType] =
+    List(BIT, BOOLEAN, INT, BIGINT, DECIMAL, DOUBLE, TIMESTAMP, DATE, TEXT, VARCHAR, BLOB)
+
+  // data types used in TPC-H tests
+  val tpchDataTypes: List[ReflectedDataType] = List(INT, DECIMAL, DATE, VARCHAR, CHAR)
+
+//  def isBits(dataType: DataType): Boolean = bits.contains(dataType)
```

Comment:
a lot of commented code, any reason to leave them here?

Reply:
These are examples of how to check the data type in tests. You may uncomment them if your tests involve the specific type.

Reply:
then we need add comments, INHO

Reply:
sure.